x-pack/plugins/observability_solution/inventory/public/hooks/use_discover_redirect.ts

```diff
@@ -5,52 +5,36 @@
  * 2.0.
  */
 import {
-  ENTITY_TYPE,
   ENTITY_DEFINITION_ID,
   ENTITY_DISPLAY_NAME,
   ENTITY_LAST_SEEN,
+  ENTITY_TYPE,
 } from '@kbn/observability-shared-plugin/common';
 import { useCallback } from 'react';
-import { type PhrasesFilter, buildPhrasesFilter } from '@kbn/es-query';
-import type { DataViewField } from '@kbn/data-views-plugin/public';
 import type { Entity, EntityColumnIds } from '../../common/entities';
 import { unflattenEntity } from '../../common/utils/unflatten_entity';
 import { useKibana } from './use_kibana';
-import { useInventoryParams } from './use_inventory_params';
-import { useInventorySearchBarContext } from '../context/inventory_search_bar_context_provider';
+import { useUnifiedSearchContext } from './use_unified_search_context';
 
 const ACTIVE_COLUMNS: EntityColumnIds[] = [ENTITY_DISPLAY_NAME, ENTITY_TYPE, ENTITY_LAST_SEEN];
 
 export const useDiscoverRedirect = () => {
   const {
     services: { share, application, entityManager },
   } = useKibana();
-  const {
-    query: { kuery, entityTypes },
-  } = useInventoryParams('/*');
 
-  const { dataView } = useInventorySearchBarContext();
+  const { dataView, searchState } = useUnifiedSearchContext();
 
   const discoverLocator = share.url.locators.get('DISCOVER_APP_LOCATOR');
 
   const getDiscoverEntitiesRedirectUrl = useCallback(
     (entity?: Entity) => {
-      const filters: PhrasesFilter[] = [];
-
-      const entityTypeField = (dataView?.getFieldByName(ENTITY_TYPE) ??
-        entity?.[ENTITY_TYPE]) as DataViewField;
-
-      if (entityTypes && entityTypeField && dataView) {
-        const entityTypeFilter = buildPhrasesFilter(entityTypeField, entityTypes, dataView);
-        filters.push(entityTypeFilter);
-      }
-
       const entityKqlFilter = entity
         ? entityManager.entityClient.asKqlFilter(unflattenEntity(entity))
         : '';
 
       const kueryWithEntityDefinitionFilters = [
-        kuery,
+        searchState.query.query,
```

Comment:
Nit: Should we name this `kuery` so it will be `searchState.query.kuery` or call it just `searchState.query`, wdyt?

Reply:
`Host` uses **query** too `_a=(dateRange:(from:now-15m,to:now),filters:!(),limit:100,panelFilters:!(),query:(language:kuery,query:''))` so I'd rather keep it consistent.